snowpack/src/config.ts

```diff
@@ -811,13 +814,24 @@ export function createConfiguration(
 
 export function loadAndValidateConfig(flags: CLIFlags, pkgManifest: any): SnowpackConfig {
   const explorerSync = cosmiconfigSync(CONFIG_NAME, {
-    // only support these 4 types of config for now
+    // only support these 5 types of config for now
     searchPlaces: [
       'package.json',
       'snowpack.config.cjs',
       'snowpack.config.js',
+      'snowpack.config.ts',
       'snowpack.config.json',
     ],
+    loaders: {
```

Comment:
strong +1 on this feature request, but I think this implementation will break any relative imports (`../` out of the temp dir is going to point somewhere else).

I'm not sure of a better way to do this though. We could create the '.snowpack.config.cjs' file alongside `snowpack.config.ts` temporarily, and then delete it when we're done. It sounds like cosmicconfig does some caching, which means this wouldn't have to happen every time. That's the best I can come up with right now

Reply:
I've re-implemented this part (see latest commit). The new way is using Node.js built-in `vm` module and execute it. Because the code is wrapped in a function, variables created in the config file won't be leaked into outer scope. And, there won't be filesystem operations.

Reply:
Now the config file will be bundled by esbuild. That means any imports will be bundled into a single file, so placing the compiled file in temp directory should be safe now.

Reply:
That's clever! It's definitely a solution to the relative imports problem but introduces a lot of extra effort with potential for slow down and bugs. I wouldn't feel comfortable unless this also came with a `console.warn("TypeScript config support is still experimental")`.

tbh I still prefer the temporary single-file method outlined in https://github.com/pikapkg/snowpack/pull/1253/files#r501834625 to bundling the entire config. A light touch for a workaround like this is valuable. And, since cosmicconfig does caching, the users should rarely notice it.

Reply:
While we can put the temporary file into current working directory,  I think bundling is necessary, because this allows to import other TS files.

Reply:
ping @FredKSchott

Reply:
still thinking about this. St the very least, we want to get the current release that's on master out first, and then revisit this post-release. Hopefully getting that out tonight/tomorrow.